### PR TITLE
Refactored analysis workflow.

### PR DESCRIPTION
**Pull request by jgallowa07**
This pull request changes several things about the `n_t` analysis workflow. 
1. Each analysis run requires a config file.
2. Plotting has been bulked up and includes the ground truth now.
3. the intermediate and Results files have been separated such that you can pass around the predictions (presumably for data analysis and plotting) without the bulky intermediate files generated as a by-product of the analysis.
4. Updated documentation. 
5. a Few parameters in the Snakefile pertaining to the individual software parameters passed - most notably generation time and bootstrap iterations.

NOTE: To run an analysis with the code in this PR, you must have the bleeding edge `msprime` installed (For ground truth coalescent rates).

### CODE REVIEW

**Comment by jgallowa07** [comment]
@andrewkern You're probably the best one to give this a review.

File: n_t/Snakefile [hunk 1]
```diff
@@ -132,25 +142,25 @@ rule sp_download:
...
     threads: 20
     run: 
-        inputs = expand(output_dir + "/{seeds}/{chrms}.trees",
+        inputs = expand(output_dir + "/Intermediate/{seeds}/{chrms}.trees",
             seeds=wildcards.seeds, chrms=chrm_list)
         runner = stairway.StairwayPlotRunner(
             workdir= output_dir + "/stairwayplot/" + wildcards.seeds+"/",
             stairway_dir=pathlib.Path.cwd() / "stairwayplot")
-        runner.ts_to_stairway(inputs, num_bootstraps=12)
+        runner.ts_to_stairway(inputs, num_bootstraps=200)
```
Comment (andrewkern) [inline n_t/Snakefile]: 200 seems like a good default, but can will pull this out to the config file as well?

File: n_t/Snakefile [hunk 2]
```diff
@@ -270,15 +280,18 @@ rule all_plot:
...
+                    pop_id=population_id)
 
 
 rule clean:
     shell:
-        "rm -rf stairwayplot \
+        f"rm -rf stairwayplot \
             stairwayplot.tar.gz \
-            "+output_dir+" \
+            {output_dir}/Intermediate \
```
Comment (andrewkern) [inline n_t/Snakefile]: this use of `Intermediate` is smart

File: n_t/msmc.py [hunk 3]
```diff
@@ -60,12 +60,12 @@ def run_msmc_estimate(input_files, output_file, msmc_exec_loc, iterations=1, nco
     """
     
     cmd = (f"{msmc_exec_loc} --fixedRecombination -o \
-        {output_file} -i {iterations} -t {ncores} {input_files}")
```
Comment (andrewkern) [inline n_t/msmc.py]: why no more `ncores`?

Comment (jgallowa07) [reply n_t/msmc.py]: Without this parameter, it uses all available cores, which is more straightforward than `-j` which specifies the number of `rules` which can be run in parallel. But I'm happy to include it.

File: n_t/Snakefile (hunk 1 above)
Comment (jgallowa07) [reply n_t/Snakefile]: Sure.

File: n_t/msmc.py (hunk 3 above)
Comment (andrewkern) [reply n_t/msmc.py]: ah definitely more straightforward that `-j`. keep it this way.

File: n_t/Snakefile (hunk 1 above)
Comment (andrewkern) [reply n_t/Snakefile]: if you make that small change and commit, let's merge

**Comment by jeromekelleher** [comment]
One minor point here: this depends on unreleased functionality in msprime, which will probably trip some people up. I'm hoping to get an msprime update out in the next couple of weeks, but stuff keeps popping up so I can't guarantee this will happen quickly. 

This is just FYI really, I'm not sure what we can do about it.

**Comment by jgallowa07** [comment]
@andrewkern This is good to merge.

**Comment by jgallowa07** [comment]
> One minor point here: this depends on unreleased functionality in msprime, which will probably trip some people up. I'm hoping to get an msprime update out in the next couple of weeks, but stuff keeps popping up so I can't guarantee this will happen quickly.
> 
> This is just FYI really, I'm not sure what we can do about it.

I'll make sure to document this, but for our purposes, I don't think depending on a bleeding edge install should be too much trouble at the moment.

**Comment by andrewkern** [comment]
it has been done

**Pull request merged**